code/modules/surgery/bodyparts/species_parts/genetics_bodyparts.dm

```diff
@@ -0,0 +1,222 @@
+#define BUNNY_HOP "bunny_hop"
+
+/datum/action/cooldown/spell/bunny_hop
+	name = "Bunny Hop"
+	desc = "Hop a distance with your bunny leg(s)! Go further the more bunny limbs you've got."
+	button_icon = 'icons/mob/actions/actions_items.dmi'
+	button_icon_state = "jetboot"
+	cooldown_time = 7 SECONDS
+	spell_requirements = NONE
+	var/mob/living/carbon/human/last_caster
+
+/datum/action/cooldown/spell/bunny_hop/cast(mob/living/cast_on)
+	. = ..()
+	last_caster = cast_on
+	var/bunny_multiplier = 0
+	var/mob/living/carbon/human/bunny = cast_on
+	for(var/obj/item/bodypart/bodypart in bunny.bodyparts)
+		if(bodypart.limb_id == BODYPART_ID_RABBIT || bodypart.limb_id == BODYPART_ID_DIGITIGRADE)
+			bunny_multiplier++
+	if(bunny.getStaminaLoss() > 0 || bunny.legcuffed) // cannot reach maximum jump if you have any stamina loss or are legcuffed(bola, bear trap, etc.)
+		bunny_multiplier = min(bunny_multiplier, 5)
+		cast_on.visible_message(span_warning("[cast_on] weakly hops with their genetically-engineered rabbit legs, hampered by their lack of stamina!"))
+		cast_on.balloon_alert_to_viewers("weakly hops")
+	else
+		cast_on.visible_message(span_warning("[cast_on] hops with their genetically-engineered rabbit legs!"))
+		cast_on.balloon_alert_to_viewers("hops")
+	playsound(cast_on, 'sound/effects/arcade_jump.ogg', 75, vary=TRUE)
+
+
+	cast_on.layer = ABOVE_MOB_LAYER
+	if(bunny_multiplier >= 6) // they have committed to the bit, so we will reward it
+		cast_on.pass_flags |= PASSTABLE|PASSGRILLE|PASSWINDOW|PASSMACHINE|PASSSTRUCTURE
+		RegisterSignal(cast_on, COMSIG_MOVABLE_MOVED, PROC_REF(break_glass))
+	else
+		cast_on.pass_flags |= PASSTABLE|PASSGRILLE|PASSMACHINE|PASSSTRUCTURE
+		RegisterSignal(cast_on, COMSIG_MOVABLE_MOVED, PROC_REF(break_grilles))
```

Comment:
https://github.com/tgstation/tgstation/pull/79698#issuecomment-1817666120

`PASSTABLE` must be handled with `passtable_on`

```suggestion
		cast_on.pass_flags |= PASSGRILLE|PASSWINDOW|PASSMACHINE|PASSSTRUCTURE
		RegisterSignal(cast_on, COMSIG_MOVABLE_MOVED, PROC_REF(break_glass))
		passtable_on(cast_on, REF(src))
	else
		cast_on.pass_flags |= PASSTABLE|PASSGRILLE|PASSMACHINE|PASSSTRUCTURE
		RegisterSignal(cast_on, COMSIG_MOVABLE_MOVED, PROC_REF(break_grilles))
		passtable_on(cast_on, REF(src))
```